src/extension/infer.rs

```diff
@@ -311,11 +291,13 @@ impl UnificationContext {
             match node_type.signature() {
                 // Input extensions are open
                 None => {
-                    self.gen_union_constraint(
-                        m_input,
-                        m_output,
-                        node_type.op_signature().extension_reqs,
-                    );
+                    let delta = node_type.op_signature().extension_reqs;
+                    let c = if delta.is_empty() {
+                        Constraint::Equal(m_input)
```

Comment:
The alternative would be to *store* this with the empty ExtensionSet, and then we would only be storing one kind of Constraint (so, a struct, not an enum - might be syntactically easier to work with!). Then we'd be testing `ExtensionSet::is_empty` instead of matching on `Constraint::Plus` vs `::Equals`...